src/main/java/magicbees/itemInventories/InventoryBeeRing.java

```diff
@@ -0,0 +1,339 @@
+            InventoryBaubles baubles = PlayerHandler.getPlayerBaubles(this.player);
+            baubles.setInventorySlotContents(itemLocatedSlot, parent);
+        } else {
+            player.inventory.setInventorySlotContents(itemLocatedSlot, parent);
+        }
+    }
+
+    // Writes the effectData field to NBT in a readable way (This and the next method could probably be condensed, but
+    // I feel that this is the most readable form I could come up with.
+    public void writeEffectNBT() {
+        NBTTagCompound compound = parent.getTagCompound();
+
+        if (compound == null) {
+            compound = new NBTTagCompound();
+            parent.setTagCompound(compound);
+        }
+
+        NBTTagCompound slotNbt1;
+        if (!compound.hasKey(KEY_EFFECT_1)) {
+            slotNbt1 = new NBTTagCompound();
+            compound.setTag(KEY_EFFECT_1, slotNbt1);
+        } else {
+            slotNbt1 = compound.getCompoundTag(KEY_EFFECT_1);
+        }
+
+        NBTTagCompound slotNbt2;
+        if (!compound.hasKey(KEY_EFFECT_2)) {
+            slotNbt2 = new NBTTagCompound();
+            compound.setTag(KEY_EFFECT_2, slotNbt2);
+        } else {
+            slotNbt2 = compound.getCompoundTag(KEY_EFFECT_2);
+        }
+
+        for (int i = 0; i < 5; i++) {
+            String slotKey = getSlotNBTKey(i);
+            try {
+                slotNbt1.setInteger(slotKey + "i", effectData[0].getInteger(i));
+                slotNbt1.setBoolean(slotKey + "b", effectData[0].getBoolean(i));
+                slotNbt1.setFloat(slotKey + "f", effectData[0].getFloat(i));
+            } catch (Exception ignored) {}
+        }
+
+        for (int i = 0; i < 5; i++) {
+            String slotKey = getSlotNBTKey(i);
+            try {
+                slotNbt2.setInteger(slotKey + "i", effectData[1].getInteger(i));
+                slotNbt2.setBoolean(slotKey + "b", effectData[1].getBoolean(i));
+                slotNbt2.setFloat(slotKey + "f", effectData[1].getFloat(i));
+            } catch (Exception ignored) {}
+        }
+
+        setInventorySlot(parent);
+    }
+
+    // Creates the correct implementation of IEffectData and pulls the data saved in the NBT into the effectData field
+    public void setEffectAndInitialize(IBee queen) {
+        effectData[0] = queen.getGenome().getEffect().validateStorage(effectData[0]);
+        effectData[1] = ((IAlleleBeeEffect) queen.getGenome().getInactiveAllele(EnumBeeChromosome.EFFECT))
+                .validateStorage(effectData[1]);
+
+        NBTTagCompound compound = parent.getTagCompound();
+
+        if (compound.hasKey(KEY_EFFECT_1)) {
+            NBTTagCompound nbtSlots1 = compound.getCompoundTag(KEY_EFFECT_1);
+
+            for (int i = 0; i < 5; i++) {
+                String slotKey = getSlotNBTKey(i);
+
+                if (nbtSlots1.hasKey(slotKey + "i")) {
+                    int effectNBTInt = nbtSlots1.getInteger(slotKey + "i");
+                    try {
```

Comment:
Why do you use try-catch? Should null check go there or what?

Reply:
effectData[0]'s implementation has private fields that can't be accessed. I have no way to get the length from these, so I need to check if I can set it. Not sure how else to do this.